test/test_distributed.py

```diff
@@ -1366,13 +1367,69 @@ def test_all_gather_cuda(self):
     @unittest.skipIf(BACKEND == "nccl", "Nccl does not support CPU tensors")
     def test_all_gather_group(self):
         group, group_id, rank = self._init_group_test()
+        print("group", group, group_id)
         self._test_all_gather_helper(group, group_id, rank)
 
     @unittest.skipIf(BACKEND == "nccl", "Nccl does not support CPU tensors")
     def test_all_gather_full_group(self):
         group, group_id, rank = self._init_full_group_test()
         self._test_all_gather_helper(group, group_id, rank)
 
+
+    def _test_all_gather_coalesced_helper(
```

Comment:
This test case only covers 3D tensors. Let's add some tests to cover more corner cases.

1. input tensors are of different dimensions, e.g., `[torch.zeros(2, 2), torch.zeros(3, 3, 3), ...]` 
0. empty input tensors, e.g., `[torch.zeros([0])]` and `[torch.zeros([0]), torch.zeros([2, 2])]`

Let's put those in separate test cases so that when it fails, we know quickly what went wrong.

Reply:
Done. 

BTW, passing only empty tensor breaks this, so we have to have at least non-empty one. I think that's a reasonable requirement. I've added it in description section.